trinity/plugins/builtin/attach/__init__.py

```diff
@@ -0,0 +1,12 @@
+from trinity.plugins.builtin.attach.plugin import (
+    AttachPlugin
+)
+
+
+def get_suited_attach_plugin() -> AttachPlugin:
+    try:
+        import IPython  # noqa: F401
```

Comment:
Maybe since this has been determined to have a side effect we can instead use the following pattern.

```python
def is_ipython_available():
    try:
        pkg_resources.get_distribution('IPython')
    except pkg_resources.DistributionNotFound:
        return False
    else:
        return True
```

I think there **might** be some weirdness in whether `DistributionNotFound` is available but that might be isolated to older python versions.